include/shared_game_state.hpp

```diff
@@ -0,0 +1,41 @@
+#ifndef SHARED_GAME_STATE_HPP_
+#define SHARED_GAME_STATE_HPP_
+#include <limits>
+#include <vector>
+#include "game_state.hpp"
+#include "model_player.hpp"
+
+namespace meow::model {
+
+struct SharedGameState {
+private:
+    std::vector<Player> players;
+    std::size_t cur_turn = std::numeric_limits<std::size_t>::max();
+
+public:
+    void set_player_list(const std::vector<std::size_t> &users);
+
+    std::size_t get_current_user_id() const;
+
+    std::size_t get_number_of_players() const;
+
+    std::size_t get_card_id_from_deck();
+
+    Player *get_player_by_player_id(std::size_t player_id);
+
+    Player *get_player_by_user_id(std::size_t user_id);
+
+    std::size_t get_player_position_by_user_id(std::size_t user_id) const;
```

Comment:
const

Reply:
not marked as const because game_state must have access to change player fields